service/history/replicatorQueueProcessor.go

```diff
@@ -441,7 +442,8 @@ func (p *replicatorQueueProcessorImpl) getTasks(
 		metrics.InstanceTag(strconv.Itoa(p.shard.GetShardID())),
 	)
 	taskGeneratedTimer := replicationScope.StartTimer(metrics.TaskLatency)
-	taskInfoList, hasMore, err := p.readTasksWithBatchSize(lastReadTaskID, p.fetchTasksBatchSize)
+	shardID := p.shard.GetShardID()
+	taskInfoList, hasMore, err := p.readTasksWithBatchSize(lastReadTaskID, p.fetchTasksBatchSize(shardID))
```

Comment:
Please keep an eye on the GC activity caused by this function is invoked very often